crux-core/src/crux/api/ICruxAPI.java

```diff
@@ -49,6 +49,16 @@
      */
     public Map<Keyword,?> document(Object contentHash);
 
+    /**
+     *  Reads a document from the document store based on its
+     *  content hash.
+     *
+     * @param contentHashSet a set of objects that can be coerced into a content
+     * hashes.
+     * @return            a map from hashable objects to the corresponding documents.
+     */
+    public Map<String,?> documents(Object contentHashSet);
```

Comment:
As you are keen on keeping this a set, this should be `Set<?>` I think. That said, I would be fine relaxing this across the board to `Collection` - while a set is conceptually correct, it's not strictly necessary.